src/test/java/com/microsoft/azure/kusto/kafka/connect/sink/Utils.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.azure.kusto.kafka.connect.sink;
+
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.Objects;
+
+public class Utils {
+    private static final Logger log = LoggerFactory.getLogger(FileWriter.class);
+
+    private Utils() {
+
+    }
+
+    public static File getCurrentWorkingDirectory() {
+        File currentDirectory = new File(Paths.get(
+                System.getProperty("java.io.tmpdir"),
+                FileWriter.class.getSimpleName(),
+                String.valueOf(Instant.now().toEpochMilli())).toString());
+        boolean opResult = restrictPermissions(currentDirectory);
+        String fullPath = currentDirectory.getAbsolutePath();
+        if (!opResult) {
+            log.warn("Setting permissions on the file {} failed", fullPath);
+        }
+        currentDirectory.deleteOnExit();
+        return currentDirectory;
+    }
+
+    public static boolean createDirectoryWithPermissions(String path) {
+        File folder = new File(FilenameUtils.normalize(path));
+        folder.deleteOnExit();
+        boolean opResult = restrictPermissions(folder);
+        if (!opResult) {
+            log.warn("Setting creating folder {} with permissions", path);
+        }
+        return folder.mkdirs();
+    }
+
+    public static boolean restrictPermissions(File file) {
+        return file.setExecutable(false) && file.setReadable(true) && file.setWritable(true);
+    }
+
+    public static int getFilesCount(String path) {
+        File folder = new File(path);
+        return Objects.requireNonNull(folder.list()).length;
```

Comment:
Can we add the string message saying "the folder list cannot be null" or something which will make the code more readable rather than just throwing the NPE because NPE would have anyways been thrown without the Objects.requireNonNull

Reply:
Is used in the assertion, so the assertion would fail